OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/ResetPasswordAlertDialogFragment.java

```diff
@@ -0,0 +1,200 @@
+package org.edx.mobile.view.dialog;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.TextInputEditText;
+import android.support.design.widget.TextInputLayout;
+import android.support.v7.app.AlertDialog;
+import android.view.View;
+import android.view.accessibility.AccessibilityEvent;
+
+import com.google.inject.Inject;
+
+import org.edx.mobile.R;
+import org.edx.mobile.authentication.LoginService;
+import org.edx.mobile.http.CallTrigger;
+import org.edx.mobile.http.ErrorHandlingCallback;
+import org.edx.mobile.model.api.ResetPasswordResponse;
+import org.edx.mobile.util.InputValidationUtil;
+import org.edx.mobile.util.NetworkUtil;
+import org.edx.mobile.util.images.ErrorUtils;
+
+import roboguice.RoboGuice;
+
+public class ResetPasswordAlertDialogFragment extends AlertDialogFragment {
+
+    public interface Listener {
+        void onResult(boolean result, @Nullable String errorMessage);
+        void omDismissed();
+    }
+
+    @Inject
+    private LoginService loginService;
+
+    protected static final String EXTRA_LOGIN_EMAIL = "login_email";
+
+    private TextInputEditText et;
+    private TextInputLayout til;
+    private Listener listener;
+
+    public static ResetPasswordAlertDialogFragment newInstance(@NonNull Context context, @Nullable String email) {
+        ResetPasswordAlertDialogFragment fragment = new ResetPasswordAlertDialogFragment();
+        Bundle args = new Bundle();
+        args.putString(ARG_TITLE, context.getResources().getString(R.string.confirm_dialog_title_help));
+        args.putString(ARG_MESSAGE, context.getResources().getString(R.string.confirm_dialog_message_help));
+        args.putString(EXTRA_LOGIN_EMAIL, email);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        AlertDialog alertDialog = (AlertDialog) super.onCreateDialog(savedInstanceState);
+        RoboGuice.injectMembers(getContext(), this);
+
+        et = new TextInputEditText(getContext());
+        til = new TextInputLayout(getContext());
+        til.addView(et);
+        til.setHint(getContext().getString(R.string.email));
+
+        int paddingWidth = Math.round(getResources().getDimension(R.dimen.alert_dialog_padding_width));
+        int paddingHeight = Math.round(getResources().getDimension(R.dimen.alert_dialog_padding_height));
+        til.setPadding(paddingWidth, paddingHeight, paddingWidth, paddingHeight);
```

Comment:
You have two views (`TextInputLayout` and `TextInputEditText`) with static attributes (hint) and padding. This would definitely be better defined in an XML layout. Views should be defined in Java code only when they have no non-default attributes (and specially not related to dimensions/layout).